ext/CMakeLists.txt

```diff
@@ -1,420 +1,435 @@
+      ${CMAKE_COMMAND}
+        -DBUILD_TYPE:STRING=${CMAKE_CFG_INTDIR}
+        -DPREFIX:PATH=${CMAKE_CURRENT_BINARY_DIR}
+        -DWORKING_DIR:PATH=${CMAKE_CURRENT_BINARY_DIR}/build/src/ProteoWizard
+        -P  ${CMAKE_CURRENT_SOURCE_DIR}/build_pwiz.cmake
+    INSTALL_COMMAND
+      ${CMAKE_CURRENT_SOURCE_DIR}/create_links.sh
+         --prefix=${CMAKE_CURRENT_BINARY_DIR}
+      COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_CURRENT_SOURCE_DIR}/patches/boost/asio.hpp
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost
+      COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${CMAKE_CURRENT_SOURCE_DIR}/patches/boost/asio
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost/asio
+      COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${CMAKE_CURRENT_BINARY_DIR}/build/src/ProteoWizard/libraries/boost_aux/boost/utility
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost/utility
+    BUILD_IN_SOURCE 1
+  )
+else (UNIX)
+  ExternalProject_Add(
+    ProteoWizard
+    ${PWIZ_SOURCE}
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/build
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND
+      ${CMAKE_COMMAND}
+        -DBUILD_32:BOOL=${BUILD_32}
+        -DBUILD_TYPE:STRING=${CMAKE_CFG_INTDIR}
+        -DPREFIX:PATH=${CMAKE_CURRENT_BINARY_DIR}
+        -DWORKING_DIR:PATH=${CMAKE_CURRENT_BINARY_DIR}/build/src/ProteoWizard
+        -P  ${CMAKE_CURRENT_SOURCE_DIR}/build_pwiz.cmake
+    INSTALL_COMMAND
+      ${CMAKE_CURRENT_SOURCE_DIR}/create_links.sh
+         --prefix=${CMAKE_CURRENT_BINARY_DIR}
+      COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_CURRENT_SOURCE_DIR}/patches/boost/asio.hpp
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost
+      COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${CMAKE_CURRENT_SOURCE_DIR}/patches/boost/asio
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost/asio
+      COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${CMAKE_CURRENT_BINARY_DIR}/build/src/ProteoWizard/libraries/boost_aux/boost/utility
+        ${CMAKE_CURRENT_BINARY_DIR}/include/boost/utility
+    BUILD_IN_SOURCE 1
+  )
+endif (WIN32 AND NOT CYGWIN)
+
+if (UNIX)
+  ExternalProject_Add(
+    percolator
+    GIT_REPOSITORY https://github.com/percolator/percolator.git
+    GIT_TAG ${PERCOLATOR_GIT_VERSION}
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/build
+    CMAKE_ARGS ${TOOLCHAIN_FILE}
+      -DCMAKE_VERBOSE_MAKEFILE=ON
+      -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}
+      -DCMAKE_PREFIX_PATH=${CMAKE_CURRENT_BINARY_DIR}
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCRUX=${CMAKE_CURRENT_SOURCE_DIR}/..
+      -DEXT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}/include
+    PATCH_COMMAND ${CMAKE_COMMAND}  -E copy
+      ${CMAKE_CURRENT_SOURCE_DIR}/patches/percolator/src/CMakeLists.txt
+      ${CMAKE_CURRENT_BINARY_DIR}/build/src/percolator/src/CMakeLists.txt
+    INSTALL_COMMAND ""
+  )
+else (UNIX)
+  ExternalProject_Add(
+    percolator
+    GIT_REPOSITORY https://github.com/percolator/percolator.git
+    GIT_TAG ${PERCOLATOR_GIT_VERSION}

```

Comment:
I can't find where `PERCOLATOR_GIT_VERSION` is assigned. I was wondering if we should define versions for the other tools in the same place.

Reply:
By default it's not assigned, and will be empty. That results in the most recent commit of the master branch being pulled. It can be assigned on the cmake command line:

cmake -DCMAKE_BUILD_TYPE=Release -DPERCOLATOR_GIT_VERSION=rel-3-03

for example.

I think your suggestion is good though. Any specifications about which build of dependency we use should be set in the top-level CMakeLists.txt file, and deliverable at the cmake command line